Week3/homework/3-step3.js

```diff
@@ -2,23 +2,39 @@
 
 // use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
-  // add your code here
-  return str;
+  let newStr = str;
+  if (num > 0) {
+    for (let i = 2; i <= num; i++) {
+      str += newStr;
+    }
+    return str;
+  }
+  else { return str = "" }
 }
 
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
 // use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
-  // add your code here
+  let newStr = str;
+  while (num > 1) {
+    str += newStr;
+    num--;
+  }
   return str;
 }
 
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 
 // use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
-  // add your code here
+  let newStr = str;
+  if (num > 1) {
```

Comment:
This need to add an extra test at the top of the `do-while` loop illustrates nicely why the `do-while` loop is not so useful. I never use it myself. But still, you need to know about it.